modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/security/DoubleSignDetectProgram.scala

```diff
@@ -0,0 +1,34 @@
+package org.tessellation.node.shared.infrastructure.security
+
+import cats.effect.Async
+import cats.effect.std.Supervisor
+import cats.syntax.all._
+
+import scala.concurrent.duration.FiniteDuration
+
+import org.tessellation.node.shared.domain.Daemon
+import org.tessellation.node.shared.domain.fork.ForkInfoStorage
+import org.tessellation.node.shared.domain.security.{DoubleSignDetect, DoubleSignDetectRunner}
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+object DoubleSignDetectProgram {
+
+  def make[F[_]: Async](storage: ForkInfoStorage[F], doubleSignValidator: DoubleSignDetect[F]): DoubleSignDetectRunner[F] =
+    new DoubleSignDetectRunner[F] {
+
+      val logger = Slf4jLogger.getLogger
+
+      def run: F[Unit] = storage.getForkInfo.flatMap {
+        _.forks.toList.traverse_ {
+          case (id, entries) =>
+            doubleSignValidator.validate(id, entries).traverse_ { detection =>
+              logger.warn(s"Found a double signing: $detection")
```

Comment:
Please use show instance wherever possible

Reply:
Changed to using the `show` instance.